base/server/src/main/java/org/dogtagpki/server/cli/SubsystemRangeGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,291 @@
+                RequestRepository.PROP_REQUEST_ID_GENERATOR,
+                RequestRepository.DEFAULT_REQUEST_ID_GENERATOR);
+        idGenerator = IDGenerator.fromString(value);
+
+        if (newGenerator == IDGenerator.RANDOM && idGenerator != IDGenerator.RANDOM) {
+            logger.debug("Remove request ranges from configuration");
+            dbConfig.remove(DatabaseConfig.MIN_REQUEST_NUMBER);
+            dbConfig.remove(DatabaseConfig.MAX_REQUEST_NUMBER);
+            dbConfig.remove(DatabaseConfig.REQUEST_INCREMENT);
+            dbConfig.remove(DatabaseConfig.REQUEST_LOW_WATER_MARK);
+            dbConfig.remove(DatabaseConfig.REQUEST_CLONE_TRANSFER_NUMBER);
+            dbConfig.remove(DatabaseConfig.REQUEST_RANGE_DN);
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_GENERATOR, newGenerator.toString());
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_LENGTH, "128");
+            return;
+        }
+        if (newGenerator == IDGenerator.LEGACY_2 && idGenerator == IDGenerator.LEGACY) {
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_GENERATOR, newGenerator.toString());
+            return;
+        }
+        throw new EBaseException("Update to " + newGenerator + " not supported");
+    }
+    
+    private void updateRanges(DatabaseConfig dbConfig, LdapBoundConnection conn, String baseDN, String rangeDN, String defaultEndSerialNumber,
+        String hostName, String securePort) throws Exception{
+
+        LDAPSearchResults ranges = conn.search(rangeDN, LDAPv3.SCOPE_SUB, "(objectClass=pkiRange)", null, false);
+
+        BigInteger lastUsedSerial = BigInteger.ZERO;
+        boolean nextRangeToUpdate = true;
+        while (ranges.hasMoreElements()) {
+            LDAPEntry entry = ranges.next();
+            String endRange = entry.getAttribute("endRange").getStringValues().nextElement();
+            String host = entry.getAttribute("host").getStringValues().nextElement();
+            String port = entry.getAttribute("securePort").getStringValues().nextElement();
+            BigInteger next = new BigInteger(endRange, 16);
+            if (lastUsedSerial.compareTo(next) < 0) {
+                lastUsedSerial = next;
+                nextRangeToUpdate = host.equals(hostName) && port.equals(securePort);
+                
+            }
+        }
+
+        if (nextRangeToUpdate) {
+            if (lastUsedSerial == BigInteger.ZERO) {
+                lastUsedSerial = new BigInteger(defaultEndSerialNumber, 16);
+            }
+            BigInteger nextSerialNumber = lastUsedSerial.add(BigInteger.ONE);
+            String serialDN = dbConfig.getSerialDN() + "," + baseDN;
+            // store nextRange as decimal
+            LDAPAttribute attrSerialNextRange = new LDAPAttribute("nextRange", nextSerialNumber.toString());
+
+            LDAPModification serialmod = new LDAPModification(LDAPModification.REPLACE, attrSerialNextRange);
+
+            conn.modify(serialDN, serialmod);
```

Comment:
The order does not matter. This update only by the CA updating last assigned range. All other CAs will not modify that range and as a result will not touch the nextRange. The order does not matter and it should not matter id the DSs have already by synced since each CA modify only its related entries.